test/distributed/_tensor/test_matrix_ops.py

```diff
@@ -120,6 +131,108 @@ def test_placement_comb(
             test_placement_comb([spec[0]], [spec[1]])
 
+    @with_comms
+    @skip_unless_torch_gpu
+    @unittest.skipIf(not PLATFORM_SUPPORTS_FP8, "torch._scaled_mm requires H100+")
+    def test_scaled_mm(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        shrd0 = Shard(0)
+        shrd1 = Shard(1)
+        repl = Replicate()
+        part = Partial()
+
+        ws = self.world_size
+        m, n, k = 64 * ws, 48 * ws, 128 * ws
+
+        t1 = torch.randn(m, k, device=self.device_type)
+        t2 = torch.randn(n, k, device=self.device_type)
+
+        for (
+            output_spec,
+            t1_spec,
+            t2_spec,
+            scale1_shape,
+            scale2_shape,
+            scale1_spec,
+            scale2_spec,
+        ) in [
+            # Tensor-wise scaling
+            # Replicated, zero-dim scale
+            (repl, repl, repl, (), (), repl, repl),
+            # Column-parallel, two-dim scale
+            (shrd1, repl, shrd0, (1, 1), (1, 1), repl, repl),
+            # Row-parallel, one-dim scale
+            (part, shrd1, shrd1, (1,), (1,), repl, repl),
+            # Row-wise scaling
+            # Replicated
+            (repl, repl, repl, (m, 1), (n, 1), repl, repl),
+            # Column-parallel
+            (shrd1, repl, shrd0, (m, 1), (n, 1), repl, shrd0),
+            # Row-parallel (which actually ends up doing sub-row-wise scaling)
+            (part, shrd1, shrd1, (m, ws), (n, ws), shrd1, shrd1),
+        ]:
+            scale1 = torch.randn(scale1_shape, device=self.device_type)
+            scale2 = torch.randn(scale2_shape, device=self.device_type)
+            t1_fp8 = tiled_mul(t1, scale1).to(torch.float8_e4m3fn)
+            t2_fp8 = tiled_mul(t2, scale2).to(torch.float8_e4m3fn)
+
+            # _scaled_mm can't do sub-row-wise scaling, hence we simulate it
+            if scale1.squeeze().ndim == 2 or scale2.squeeze().ndim == 2:
+                full_ref_res = torch.stack(
+                    [
+                        torch._scaled_mm(
+                            st1,
+                            st2.t(),
+                            scale_a=sscale1.contiguous(),
+                            scale_b=sscale2.contiguous().t(),
+                            out_dtype=torch.bfloat16,
+                        )
+                        for st1, st2, sscale1, sscale2 in zip(
+                            t1_fp8.tensor_split(ws, dim=-1),
+                            t2_fp8.tensor_split(ws, dim=-1),
+                            scale1.tensor_split(ws, dim=-1),
+                            scale2.tensor_split(ws, dim=-1),
+                        )
+                    ],
+                    dim=0,
+                ).sum(dim=0)
+            else:
+                full_ref_res = torch._scaled_mm(
+                    t1_fp8,
+                    t2_fp8.t(),
+                    scale_a=scale1,
+                    scale_b=scale2.t(),
```

Comment:
should this be `1 / scale1`? Can you comment a bit on what the original values, scaled values, and scales are?

Reply:
I answered above.